test/jit/test_dtype_analysis.py

```diff
@@ -164,10 +177,59 @@ def test_binary_scalar(self):
                 scalar_type = in_dtypes[1]
 
                 if scalar_type == float32:
+
                     def add(x, y: float):
                         return x + y
+
                 else:
+
                     def add(x, y: int):
                         return x + y
 
                 self.assert_dtype_equal(add, in_shapes, in_dtypes)
+
+    def test_custom_rules(self):
+        # Test some of the ops that are not covered by Metatensors
+
+        # Note that unlike the Conv2d module, the function conv2d
+        # does not take dtype/device arguments.
+
+        def conv2d_fn(input, weight, bias):
+            return torch.nn.functional.conv2d(input, weight, bias)
+
+        def adaptive_avg_pool2d_fn(input, output_size: Tuple[int]):
+            return torch._C._nn.adaptive_avg_pool2d(input, output_size)
+
+        for fn, inputs_fn in (
+            (conv2d_fn, sample_inputs_conv2d),
+            (adaptive_avg_pool2d_fn, sample_inputs_adaptive_avg_pool2d),
+        ):
+            for dtype in (torch.int8, torch.float64):
```

Comment:
how did you decide on these two types?

Reply:
Just seemed like two types that would be commonly used, and I wanted one int, and one float type